### ISSUE TITLE
`cache_shuffle_num` changes epoch length

### ISSUE THREAD

**Issue by feffy380** [tdrussell/diffusion-pipe#384]
The dataloader seems to include shuffled captions when determining dataset length. This results in epochs being longer than expected.
For example, `cache_shuffle_num = 10` changed my epoch length from 52 to 520, which is undesirable

**Comment by tdrussell** [comment]
This is due to how it's implemented, and there's not really an easy way to change it without a bunch of hacks.

When you shuffle captions multiple times, it literally duplicates the dataset that many times. It is caching text embeddings, so it will compute 10x the text embeddings and store 10x the text embeddings on disk. The whole dataset acts like it is 10x larger. It's basically the same as num_repeats, which is also effectively making the dataset larger.

If one epoch is getting too large, you can configure the eval and save intervals in terms of steps instead of epochs.

**Comment by feffy380** [comment]
The new `max_steps` setting is a suitable alternative for maintaining a fixed training length 👍
